### PR TITLE
Adds user activity tracking via Matomo

### LINKED ISSUE #1566: Add Javascript user activity tracking via Matomo
We can start the process of testing Matomo by adding simple JS tags.

We can consider this Django app: https://github.com/mikbuch/django-matomo

### PR DESCRIPTION
**Pull request by Alihassanc5**
Closes #1566 

- The feature adds tracking of user activity using Matomo

### CODE REVIEW

File: requirements.dev.txt [hunk 1]
```diff
@@ -7,4 +7,5 @@ django-test-plus>=1.4.0
 coverage
 selenium==4.7.2
 factory-boy==3.2.*
-wagtail-factories==4.0.*
\ No newline at end of file
+wagtail-factories==4.0.*
+django-matomo
```
Comment (sheralim012) [inline requirements.dev.txt:11]: Specify version

**Review by sheralim012 — CHANGES_REQUESTED** [review]

**Comment by cbunicef** [comment]
Would it be possible to separately specify URLs for global and local Matomo tracking?

**Comment by sheralim012** [comment]
> Would it be possible to separately specify URLs for global and local Matomo tracking?

It's doable but what's the use case behind this?

**Comment by cbunicef** [comment]
> > Would it be possible to separately specify URLs for global and local Matomo tracking?
> 
> It's doable but what's the use case behind this?

For example, if we have an IoGT platform deployed by a UNICEF partner on their servers. They might use Matomo (local, cloud etc) to track locally, and we might want to use our global Matomo to also track globally.

We never needed to consider multiple systems for Google Analytics because there was only one system available, but since local deployments are possible with Matomo it would be something I could see us using.

This isn't urgent, but it seemed that since the team was currently working on this PR it would be faster now than digging into it again at a later date.

**Review by sheralim012 — CHANGES_REQUESTED** [review]

**Comment by cbunicef** [comment]
Please add a feature flag for this, while it's in development.

File: requirements.dev.txt [hunk 2]
```diff
@@ -7,4 +7,5 @@ django-test-plus>=1.4.0
 coverage
 selenium==4.7.2
 factory-boy==3.2.*
-wagtail-factories==4.0.*
\ No newline at end of file
+wagtail-factories==4.0.*
+django-matomo==0.1.6
```
Comment (istride) [inline requirements.dev.txt:11]: django-matomo cannot be a dev-only dependency

**Review by istride — CHANGES_REQUESTED** [review]

**Review by istride — APPROVED** [review]

**Pull request merged**